compiler/typecheck/TcMatches.hs

```diff
@@ -569,7 +569,7 @@ tcMcStmt ctxt (BindStmt _ pat rhs bind_op fail_op) res_ty thing_inside
            -- (>>=) :: rhs_ty -> (pat_ty -> new_res_ty) -> res_ty
   = do  { ((rhs', pat_mult, pat', thing, new_res_ty), bind_op')
             <- tcSyntaxOp MCompOrigin bind_op
-                          [SynRho, SynFun SynAnyMult SynAny SynRho] res_ty $
+                          [SynRho, SynFun SynAny SynRho] res_ty $
```

Comment:
That's worrying. We still need an extra argument for `SynFun`,  don't we? Otherwise we're breaking the non-omega things. Doesn't it break tests?

Reply:
I also removed the argument to `SynFun`. Tests pass.

Reply:
What do you do for the multiplicity, always return it?

Reply:
I'm treating everything as `SynAnyMult`. The only place in code that used `SynMult` is dead code.

Reply:
I see.

There is a comment somewhere specifying the behaviour of these templates, and what we expect back in the return list. Could you try and locate it, then fix it to your new specification?